federation/pkg/federation-controller/util/eventsink/eventsink.go

```diff
@@ -37,14 +44,80 @@ func NewFederatedEventSink(clientset fedclientset.Interface) *FederatedEventSink
 	}
 }
 
-func (fes *FederatedEventSink) Create(event *api_v1.Event) (*api_v1.Event, error) {
-	return fes.clientset.Core().Events(event.Namespace).Create(event)
+// TODO this is uses a reflection conversion path and is very expensive.  federation should update to use client-go
```

Comment:
@kubernetes/sig-federation-misc This is some fallout from making client-go authoritative.  This pull moves the last package standing, but there will be some performance impact here for you.  You should plumb the native types through and avoid the conversion.